src/Adapter/CartRule/CommandHandler/AddCartRuleHandler.php

```diff
@@ -107,8 +109,15 @@ private function fillCartRuleConditionsFromCommandData(CartRule $cartRule, AddCa
     {
         $cartRule->id_customer = null !== $command->getCustomerId() ? $command->getCustomerId()->getValue() : null;
 
-        $cartRule->date_from = $command->getValidFrom()->format('Y-m-d H:i:s');
-        $cartRule->date_to = $command->getValidTo()->format('Y-m-d H:i:s');
+        if (null === $command->getValidFrom() || null === $command->getValidTo()) {
+            $now = new DateTimeImmutable();
```

Comment:
Shouldn't we try to use the available values as much as possible?

```suggestion
            $from = $command->getValidFrom() ?: new DateTimeImmutable();
            $to = $command->getValidTo() ?: $from->modify('+1 month');
```

Reply:
command setter allows only setting the date range, so if either one of them null it means both of them are (probably it is a tiny flaw like - we are missing some date range VO in theory or so...?) or should I just allow to set $from and $to separately and take care of them here as you suggest? :thinking:

Reply:
Ah ok, I didn't notice the associated setter forces to have both or neither!! Sorry. All good for me then